lib/profileReport.js

```diff
@@ -92,86 +143,66 @@ function collectBandwidth(bytes, path, direction) {
   var node = direction[path];
   node.times += 1;
   node.bytes += bytes;
-}
+};
 
-function collect(data, path, bytes, speed, direction) {
-  collectSpeed(data, path, speed);
-  collectBandwidth(bytes, path, direction);
-}
+ProfileReport.prototype.collect = function(data, path, bytes, speed, direction) {
+  this.collectSpeed(data, path, speed);
+  this.collectBandwidth(bytes, path, direction);
+};
 
-function processOperation(data, state) {
-  var path = pathString(data.path);
-  state.opCount++;
+ProfileReport.prototype.processOperation = function(data) {
+  if (!this.state.startTime) {
+    this.state.startTime = data.timestamp;
+  }
+  this.state.endTime = data.timestamp;
+  var path = ProfileReport.pathString(data.path);
+  this.state.opCount++;
   switch (data.name) {
   case 'concurrent-connect':
     break;
   case 'concurrent-disconnect':
     break;
   case 'realtime-read':
-    collect(data, path, data.bytes, state.readSpeed, state.outband);
+    this.collect(data, path, data.bytes, this.state.readSpeed, this.state.outband);
```

Comment:
If the profile report is now a class, do you need to pass readSpeed and outband? Or can you just call `this.state.readSpeed` and `this.state.outband` from the `collect` function directly? Or have two methods: `collectRead` and `collectWrite`?